vars/promoteArtifacts.groovy

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+void call(Map args = [:]) {
+    git url: 'https://github.com/opensearch-project/opensearch-build.git', branch: 'main'
+    def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+    String manifest = args.manifest ?: "manifests/${INPUT_MANIFEST}"
+    def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifest))
+    String filename = inputManifest.build.getFilename()
+    String version = inputManifest.build.version
+
+    def artifactPath = "${DISTRIBUTION_JOB_NAME}/${version}/${DISTRIBUTION_BUILD_NUMBER}/${DISTRIBUTION_PLATFORM}/${DISTRIBUTION_ARCHITECTURE}"
```

Comment:
Are we planning to run this differently for each platform/architecture? Like twice, once for x64 and once for arm64?

Reply:
yes, and that is ok because you will have time that you dont have both arch for a platform.
e.g. no arm64 for windows release only x64.